kmip/services/kmip_client.py

```diff
@@ -217,6 +217,7 @@ def open(self):
             suppress_ragged_eofs=self.suppress_ragged_eofs)
         self.protocol = KMIPProtocol(self.socket)
 
+        self.socket.settimeout(self.timeout)
```

Comment:
Peter,

We have two notes to consider i believe, one is connection time out and another is socket time out. Connection time out especially should work when establishing a connection to kmip server. Another is socket timeout, used for kmip operations where kmip operations can perform especially batch operations EX: Kmip protocol when used for data encryption depending upon the batch operations supported and data payload, can still be a blocking socket call i believe. I mean once the connection is established, make the socket a blocking once again. Even if user provides timeout while creating a kmipproxy instance, he still with this implementation can provide only one timeout.